client/app/pages/queries/view.js

```diff
@@ -278,6 +279,26 @@ function QueryViewCtrl(
     AlertDialog.open(title, message, confirm).then(archive);
   };
 
+  $scope.unarchiveQuery = () => {
+    const queryId = $scope.query.id;
+    const url = `api/queries/${queryId}/unarchive`;
+    const unarchive = () => {
+      Events.record('unarhive', 'queries', queryId);
+      $http.patch(url).success(() => {
+        $scope.query.is_archived = false;
+      }).error(() => {
+        toastr.error('Query could not be unarchived.');
+      });
+    };
+
+    const title = 'Unarchive Query';
+    const message =
+      'Are you sure you want to unarchive this query?';
+    const confirm = { class: 'btn-warning', title: 'Unarchive' };
+
+    AlertDialog.open(title, message, confirm).then(unarchive);
```

Comment:
I'm not sure confirmation is needed here... it's not that destructive and easy to revert.

Reply:
I agree with you. I'll fix it.